test/test_utils.sh

```diff
@@ -425,55 +431,89 @@ dump_logs() {
     dump_function_pod_logs $ns $fns
     dump_builder_pod_logs $bns
     dump_fission_crds
+    travis_fold_end dump_logs
 }
 
-log() {
-    echo `date +%Y/%m/%d:%H:%M:%S`" $1"
-}
-
-export -f log
 export FAILURES=0
 
 run_all_tests() {
     id=$1
 
     export FISSION_NAMESPACE=f-$id
     export FUNCTION_NAMESPACE=f-func-$id
-
-    test_files=$(find $ROOT/test/tests -iname 'test_*.sh')
-
-    for file in $test_files
-    do
-    run_test ${file}
+    export PYTHON_RUNTIME_IMAGE=gcr.io/fission-ci/python-env:test
+    export PYTHON_BUILDER_IMAGE=gcr.io/fission-ci/python-env-builder:test
+    export GO_RUNTIME_IMAGE=gcr.io/fission-ci/go-env:test
+    export GO_BUILDER_IMAGE=gcr.io/fission-ci/go-env-builder:test
+    export JVM_RUNTIME_IMAGE=gcr.io/fission-ci/jvm-env:test
+    export JVM_BUILDER_IMAGE=gcr.io/fission-ci/jvm-env-builder:test
+
+    set +e
+    export TIMEOUT=900  # 15 minutes per test
+
+    # run tests without newdeploy in parallel.
+    export JOBS=6
+    $ROOT/test/run_test.sh \
+        $ROOT/test/tests/mqtrigger/kafka/test_kafka.sh \
+        $ROOT/test/tests/mqtrigger/nats/test_mqtrigger.sh \
+        $ROOT/test/tests/mqtrigger/nats/test_mqtrigger_error.sh \
+        $ROOT/test/tests/recordreplay/test_record_greetings.sh \
+        $ROOT/test/tests/recordreplay/test_record_rv.sh \
+        $ROOT/test/tests/recordreplay/test_recorder_update.sh \
+        $ROOT/test/tests/test_annotations.sh \
+        $ROOT/test/tests/test_archive_pruner.sh \
+        $ROOT/test/tests/test_backend_poolmgr.sh \
+        $ROOT/test/tests/test_buildermgr.sh \
+        $ROOT/test/tests/test_canary.sh \
+        $ROOT/test/tests/test_env_vars.sh \
+        $ROOT/test/tests/test_fn_update/test_idle_objects_reaper.sh \
+        $ROOT/test/tests/test_function_test/test_fn_test.sh \
+        $ROOT/test/tests/test_function_update.sh \
+        $ROOT/test/tests/test_ingress.sh \
+        $ROOT/test/tests/test_internal_routes.sh \
+        $ROOT/test/tests/test_logging/test_function_logs.sh \
+        $ROOT/test/tests/test_node_hello_http.sh \
+        $ROOT/test/tests/test_package_command.sh \
+        $ROOT/test/tests/test_pass.sh \
+        $ROOT/test/tests/test_router_cache_invalidation.sh \
+        $ROOT/test/tests/test_specs/test_spec.sh \
+        $ROOT/test/tests/test_specs/test_spec_multifile.sh
+    FAILURES=$?
+
+    # FIXME: run tests with newdeploy one by one.
+    export JOBS=1
```

Comment:
Minor: It would be nice to create an issue to separately investigate why these tests fail in parallel

Reply:
Sure. I'll create one.